util.js

```diff
@@ -0,0 +1,302 @@
+      fs.renameSync(`${oldPath}${dir}`, `${newPath}${dir}`);
+    }
+}
+
+/**
+ * This function performs the following steps:
+ * 1. Creates a Python virtual environment.
+ * 2. Installs the specified version of Tutor.
+ * 3. Installs Tutor Pearson Plugin.
+ * @async
+ * @param {string} tutorVersion - The version of Tutor to install.
+ * @param {string} tutorPearsonPluginBranch - The branch of the Pearson plugin for Tutor.
+ * @param {string} ghAccessToken - The GitHub access token for private repositories or installations.
+ * @param {Object} options - Exec options like shell, listeners etc.
+ * @throws Will throw an error if any step in the environment preparation fails.
+ * @example ->  await prepareEnvironment('3.12.0', 'https://example.com/pluginUrl', 'myGithubToken', options);
+ */
+async function prepareEnvironment(tutorVersion, tutorPearsonPluginBranch = 'main', ghAccessToken, options) {
+    try {
+      // Create and activate virtualenv
+      await exec.exec('python3', ['-m', 'venv', 'venv'], options);
+      await execInVenv('pip', ['install', `tutor==${tutorVersion}`], options);
+      await execInVenv('pip', ['install', '-e', `git+https://${ghAccessToken}@github.com/Pearson-Advance/tutor-pearson-plugin.git@${tutorPearsonPluginBranch}#egg=tutor-pearson-plugin`], options)
+      core.info('Succesfully installed Tutor Pearson Plugin');
+      
+    } catch (error) {
+      core.error(`Environment preparation failed: ${error.message}`);
+      throw error;
+    }
+} 
+
+/**
+ * Executes a given command within a Python virtual environment.
+ * @async
+ * @param {string} command - The command to execute.
+ * @param {Array<string>} args - The list of arguments for the command.
+ * @param {Object} options
+ * @throws Will throw an error if the command fails to execute.
+ * @example -> execInVenv('pip', ['-m','install', 'package_name']);
+ */
+async function execInVenv(command, args, options) {
+    const venvCommand = `venv/bin/${command}`;
+    await exec.exec(venvCommand, args, options);
+}
+
+/**
+ * Installs Tutor plugins from provided sources if any.
+ * @async
+ * @param {Array<string>} tutorPluginSources - An array of Tutor plugin source URLs or package names.
+ * @throws {Error} Throws an error if installation of any plugin fails.
+ * @returns {void} Returns nothing; side effect is the installation of plugins.
+ */
+async function installTutorPlugins(tutorPluginSources, options) {
+    if (!Array.isArray(tutorPluginSources) || tutorPluginSources.length === 0) {
+      core.warning('No valid Tutor plugin sources provided. Skipping plugin installation.');
+      return;
+    }
+
+    core.info('Installing Tutor plugins.');
+    for (const sourceUrl of tutorPluginSources) {
+      try {
+          core.info(`Attempting to install ${sourceUrl}`);
+          const pipArgs = sourceUrl.startsWith('git+') ? ['install', '-e', sourceUrl] : ['install', sourceUrl];
+          await execInVenv('pip', pipArgs, options);
```

Comment:
Variable used once. @Serafin-dev 
```suggestion
          await execInVenv(
              'pip',
              sourceUrl.startsWith('git+') ? ['install', '-e', sourceUrl] : ['install', sourceUrl];,
              options,
          );
```